src/cpp/rtps/security/SecurityManager.cpp

```diff
@@ -123,12 +124,95 @@ bool SecurityManager::init(
     security_activated = false;
 
     SecurityException exception;
+
     domain_id_ = participant_->getRTPSParticipantAttributes().builtin.domainId;
 
+    const PropertyPolicy log_properties = PropertyPolicyHelper::get_properties_with_prefix(
+          participant_->getRTPSParticipantAttributes().properties,
+          "dds.sec.log.builtin.DDS_LogTopic.");
+
+    // length(log_properties) == 0 considered as logging disable.
+    if (PropertyPolicyHelper::length(log_properties) > 0)
+    {
+      logging_plugin_ = factory_.create_logging_plugin(participant_properties);
+
+      if (logging_plugin_ != nullptr)
+      {
+        LogOptions log_options;
+        log_options.distribute = true; //TODO(artivis): defaults to true until log to file is implemented.
+        log_options.log_level = LoggingLevel::ERROR_LEVEL;
+        log_options.log_file = "";
+
+        const auto init_logging_fail = [&](){
+          logError(SECURITY, "Logging plugin not configured. Security logging will be disabled. ("
+                   << exception.what() << ").");
+          delete logging_plugin_;
+          logging_plugin_ = nullptr;
+          return false;
+        };
+
+        const std::string* const distribute = PropertyPolicyHelper::find_property(log_properties, "distribute");
+        if (distribute != nullptr)
+        {
+          if (!distribute->compare("true") || !distribute->compare("1"))
+          {
+            log_options.distribute = true;
+          }
+          else if (!distribute->compare("false") || !distribute->compare("0"))
+          {
+            log_options.distribute = false;
+          }
+          else
+          {
+            exception = SecurityException("Unknown value '" + *distribute + "' for LogOptions::distribute.");
+            return init_logging_fail();
+          }
+        }
+
+        const std::string* const log_level = PropertyPolicyHelper::find_property(log_properties, "logging_level");
+        if (log_level != nullptr)
+        {
+          if (!string_to_LogLevel(*log_level, log_options.log_level, exception))
+          {
+            return init_logging_fail();
+          }
+        }
+
+        const std::string* const log_file = PropertyPolicyHelper::find_property(log_properties, "log_file");
+        if (log_file != nullptr)
+        {
+          log_options.log_file = *log_file;
+          //TODO(artivis): Check path
+          logInfo(SECURITY, "Logging to file is not yet implemented.");
```

Comment:
Oh, maybe you haven't pushed these parts eh?

Reply:
This is yet to be cleaned up. The path is somewhat validated by the file stream successfully opening or not.